examples/Backtesting/transfer_learning.py

```diff
@@ -0,0 +1,142 @@
+# We use them to create a lookup table after creating the search space and the objective.
+three_hump_camel = botorch_function_wrapper(ThreeHumpCamel())
+six_hump_camel = botorch_function_wrapper(SixHumpCamel())
+
+#### Creating the searchspace and the objective
+
+# The parameter `POINTS_PER_DIM` controls the number of points per dimension.
+# Note that the searchspace will have `POINTS_PER_DIM**2` many points.
+
+POINTS_PER_DIM = 10
+
+# We define two numerical discrete parameters, as well as a ``TaskParameter``.
+# This parameter contains the information about the different tasks that we have in this
+# transfer learning example.
+x1 = NumericalDiscreteParameter(
+    name="x_1", values=list(np.linspace(-2, 2, POINTS_PER_DIM))
+)
+x2 = NumericalDiscreteParameter(
+    name="x_2", values=list(np.linspace(-2, 2, POINTS_PER_DIM))
+)
+task_param = TaskParameter(name="Function", values=("ThreeHump", "SixHump"))
+parameters = [x1, x2, task_param]
+
+# We now create the searchspace, objective and campaign.
+
+searchspace = SearchSpace.from_product(parameters=parameters)
+objective = Objective(
+    mode="SINGLE", targets=[NumericalTarget(name="Target", mode="MIN")]
+)
+campaign = Campaign(searchspace=searchspace, objective=objective)
+
+#### Generating the lookup table
+
+# The lookup table for the parameters and their values need to contain one column per
+# parameter, including one for the ``TaskParameter`` and one column for the target value.
+# The table is created by constructing one dataframe per task and concatenating them.
+
+x = x1.values
+y = x2.values
+xx, yy = np.meshgrid(x, y)
+
+
+lookup_3hc = pd.DataFrame({"x_1": xx.flatten(), "x_2": yy.flatten()})
+lookup_3hc["Target"] = lookup_3hc.apply(three_hump_camel, axis=1)
+lookup_3hc["Function"] = "ThreeHump"
+
+lookup_6hc = pd.DataFrame({"x_1": xx.flatten(), "x_2": yy.flatten()})
+lookup_6hc["Target"] = lookup_6hc.apply(six_hump_camel, axis=1)
+lookup_6hc["Function"] = "SixHump"
+
+lookup = pd.concat([lookup_3hc, lookup_6hc], ignore_index=True)
+
+# We print the first ten rows to show how the table looks like
+print(lookup.head(10))
+
+#### Performing the simulation loop
+
+# We can now use the `simulate_transfer_learning` function to simulate a full experiment.
+
+results_with_tf = simulate_transfer_learning(
+    campaign,
+    lookup,
+    batch_quantity=1,
+    n_doe_iterations=N_DOE_ITERATIONS,
+    n_mc_iterations=N_MC_ITERATIONS,
+)
+
+# To showcase the improvement, we also optimize the same tasks independently and use
+# the `simulate_scenarios` function.
+
+parameters_without_tl = [x1, x2]
+campaign_without_tl_3hc = Campaign(
+    searchspace=SearchSpace.from_product([x1, x2]), objective=objective
+)
+campaign_without_tl_6hc = Campaign(
+    searchspace=SearchSpace.from_product([x1, x2]), objective=objective
+)
+
+scenarios_3hc = {"ThreeHump no TL": campaign_without_tl_3hc}
+scenarios_6hc = {"SixHump no TL": campaign_without_tl_6hc}
+
```

Comment:
could be simplified by making a single dict and a single call to `simulate_scenarios`

Reply:
Are you sure? They need different lookup functions, hence different calls to `simulate_scenarios`

Reply:
they can share a lookup as long as the campaigns are configured correctly (which then would indeed pose a usecase for `active_values`)

Reply:
Ok, but then there is no need for having a dedicated backtesting and a dedicated active_values example, right? Then these would probably collapse into one example

Reply:
its still mentioned in the userguide so no need to restrict this example here

Reply:
But `simulate_transfer_learning` only accepts a single campaign, and not a dict, so this would require an outer loop in that case, is this better?

Reply:
I have tried to make the code better to read and to understand by using a `for` loop, since I did not manage to make it run using any of the simulation functions.

Reply:
? that should be straighforward

- Tl parameter should not be excluded from searchspace but instead have `active_values='name_of_the_particular_function'`
- This will result in 3 independen campaign objects
- These can be simulated with `simulate_scenarios`
- If you use a dataframe as lookup then they can share one single dataframe as lookup

Id say its not really acceptable to code your own look as this example is about our backtest utilities

Reply:
This was now changed, please resolve if the current variant works.